hal/src/rtl872x/deviceid_hal.cpp

```diff
@@ -165,6 +150,28 @@ int hal_get_device_hw_model(uint32_t* model, uint32_t* variant, void* reserved)
     return SYSTEM_ERROR_NONE;
 }
 
+int hal_get_device_hw_info(hal_device_hw_info* info, void* reserved) {
+    CHECK_TRUE(info, SYSTEM_ERROR_INVALID_ARGUMENT);
+    // HW Data format: | NCP_ID (LSB) | HW_VERSION | HW Feature Flags |
+    //                 |    byte 0    |   byte 1   |    byte 2/3      |
+    uint8_t hw_data[4] = {};
+    CHECK(readLogicalEfuse(HARDWARE_DATA_OFFSET, (uint8_t*)hw_data, sizeof(hw_data)));
+    CHECK(hal_get_device_hw_model(&info->model, &info->variant, nullptr));
+    info->revision = hw_data[1];
+    if (info->revision == 0xff) {
+        info->revision = 0xffffffff;
+    }
+    info->features = ((uint32_t)hw_data[3] << 8) | (uint32_t)hw_data[2];
+    memset(info->ncp, 0xff, sizeof(info->ncp));
+    for (int i = 0; i < HAL_PLATFORM_NCP_COUNT; i++) {
```

Comment:
For `trackerm` we are going to not count the P2 wifi as one of the NCPs? I see its defined as only having the Quectel modem here
```
#define HAL_PLATFORM_NCP_COUNT (1)
```

Reply:
Not necessarily. It's one of the potentially components that can be updated, so it doesn't need to be in the list of NCPs.

Reply:
Ok, we only have one byte in efuse/otp for NCPID, so this should be the primary NCP right? So for trackerm this should be the appropriate Quectel NCPID right? 
I guess this also begs the question of what happens if/when we have a product with two updateable/"primary" NCPs, but I guess thats not really relevant (or likely?) right now

Reply:
See latest updates. This should probably be correct for all supported platforms.